qemu/tests/qemu_guest_agent.py

```diff
@@ -294,6 +301,34 @@ def gagent_check_install_uninstall(self, test, params, env):
         session.close()
 
     @error_context.context_aware
+    def gagent_check_stop_start(self, test, params, env):
+        """
+        Repeat stop/restart qemu-ga service in guest.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        repeats = int(params.get("repeat_times", 1))
+        logging.info("Repeat stop/restart qemu-ga service for %s times" % repeats)
+
+        if not self.vm:
+            self.vm = self.env.get_vm(params["main_vm"])
+            self.vm.verify_alive()
+        session = self._get_session(params, self.vm)
+        for i in xrange(repeats):
+            error_context.context("Repeat: %s/%s" % (i + 1, repeats),
+                                  logging.info)
+            self.gagent_stop(session, self.vm)
```

Comment:
Is there some logic can make sure the ga service is running first of all?

Reply:
yes, for each gagent test case, it will run setup() function. And there is service start action  in setup() funcion.